src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -360,28 +365,27 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash {
             if (currentBeamVehicle.isBEV | currentBeamVehicle.isPHEV) {
               stall.chargingPointType match {
                 case Some(_) =>
-                  handleStartCharging(tick, currentBeamVehicle)
-                  if (ChargingPointType.getChargingPointInstalledPowerInKw(
-                        currentBeamVehicle.stall.get.chargingPointType.get
-                      ) > 20.0) {
-                    val (sessionDuration, energyDelivered) =
-                      currentBeamVehicle.refuelingSessionDurationAndEnergyInJoules()
-                    log.debug(
-                      "scheduling EndRefuelSessionTrigger at {} with {} J to be delivered, triggerId: {}",
-                      tick + sessionDuration.toInt,
-                      energyDelivered,
-                      triggerId
-                    )
-                    scheduler ! ScheduleTrigger(
-                      EndRefuelSessionTrigger(
-                        tick + sessionDuration.toInt,
-                        tick,
-                        energyDelivered,
-                        currentBeamVehicle
-                      ),
-                      self
-                    )
+                  log.debug("Sending ChargingPlugRequest to chargingNetworkManager at {}", tick)
+                  try {
+                    Await
+                      .result(
+                        chargingNetworkManager ? ChargingPlugRequest(
+                          tick,
+                          currentBeamVehicle,
+                          ChargingNetworkManager.defaultVehicleManager
```

Comment:
What is the purpose of sending a vehicle network manager? Wouldn't the manager being sent to be the one to use?

Reply:
there are different networks within the ChargingNetworkManager. But we will be handling this differently using traits.